src/ceph.in

```diff
@@ -712,14 +712,14 @@ def ping_monitor(cluster_handle, name, timeout):
     if mon_id == '*':
         run_in_thread(cluster_handle.connect, timeout=timeout)
         for m in monids():
-            s = run_in_thread(cluster_handle.ping_monitor, m)
+            s = run_in_thread(cluster_handle.ping_monitor, m, timeout=timeout)
```

Comment:
i think, timeout is specified by `--connect-timeout`, which is the timeout for *connecting* to the cluster. not for pinging a certain monitor. if user pings a certain monitor, we need to apply this setting to `run_in_thread()`.

Reply:
I agree on this. Thinking on this line, I propose the following patch instead of current one -

```
@@ -709,8 +709,8 @@ def ping_monitor(cluster_handle, name, timeout):
         return 1
 
     mon_id = name[len('mon.'):]
+    run_in_thread(cluster_handle.connect, timeout=timeout)
     if mon_id == '*':
-        run_in_thread(cluster_handle.connect, timeout=timeout)
         for m in monids():
             s = run_in_thread(cluster_handle.ping_monitor, m)
             if s is None:
@@ -718,7 +718,7 @@ def ping_monitor(cluster_handle, name, timeout):
             else:
                 print("mon.{0}".format(m) + '\n' + s)
     else:
-            s = run_in_thread(cluster_handle.ping_monitor, mon_id, timeout=timeout)
+            s = run_in_thread(cluster_handle.ping_monitor, mon_id)
             print(s)
     return 0
```
It would be a mistake to add timeout to the ping operation when pinging  a single monitor. Thus, we need to move statement that connects the cluster before the conditional that distinguishes between pinging all monitors and a single monitor.

Reply:
i don't think it's necessary to "connect" to a certain monitor for pinging it.

Reply:
I tested it. Connecting, indeed, is not required. So, are we drop the connect statement altogether? This would imply that `--connect-timeout` and this PR are redundant.

Reply:
no, `monids()` requires a *connected* `cluster_handle`.

Reply:
Yep. I made a mistake while testing. I didn't know that the bin/ceph gets replaced everytime I run `vstart.sh`.

If timeout is only for connecting to the cluster, isn't this commit futile? Why add timeout to one function call and not to the other, since both  are only pinging?

Reply:
@tchaikov Should I proceed to close this PR?

Reply:
yeah, please.